modules/objdetect/perf/perf_qrcode_pipeline.cpp

```diff
@@ -0,0 +1,58 @@
+#include "perf_precomp.hpp"
+
+namespace opencv_test
+{
+namespace
+{
+
+typedef ::perf::TestBaseWithParam< std::string > Perf_Objdetect_QRCode_Detect;
+PERF_TEST_P(Perf_Objdetect_QRCode_Detect,
+             qrcode_pipeline, ::testing::Values(
+               "version_1_down.jpg", "version_1_left.jpg", "version_1_right.jpg", "version_1_up.jpg", "version_1_top.jpg",
+               "version_5_down.jpg", "version_5_left.jpg", "version_5_right.jpg", "version_5_up.jpg", "version_5_top.jpg",
+               "russian.jpg", "kanji.jpg", "link_github_ocv.jpg", "link_ocv.jpg", "link_wiki_cv.jpg"
+             ))
+{
+    const std::string name_current_image = GetParam();
+    const std::string root = "cv/qrcode/";
+
+    std::string image_path = findDataFile(root + name_current_image);
+    Mat src = imread(image_path, IMREAD_GRAYSCALE), straight_barcode;
+    ASSERT_FALSE(src.empty()) << "Can't read image: " << image_path;
+
+    std::vector< Point > corners;
+    std::string decoded_info;
+    TEST_CYCLE() ASSERT_TRUE(detectQRCode(src, corners));
+    SANITY_CHECK(corners);
+}
+
+#ifdef HAVE_QUIRC
+typedef ::perf::TestBaseWithParam< std::string > Perf_Objdetect_QRCode_Decode;
+PERF_TEST_P(Perf_Objdetect_QRCode_Decode,
+             qrcode_pipeline, ::testing::Values(
+               "version_1_down.jpg", "version_1_left.jpg", "version_1_right.jpg", "version_1_up.jpg", "version_1_top.jpg",
+               "version_5_down.jpg", "version_5_left.jpg", "version_5_right.jpg", "version_5_up.jpg", "version_5_top.jpg",
+               "russian.jpg", "kanji.jpg", "link_github_ocv.jpg", "link_ocv.jpg", "link_wiki_cv.jpg"
```

Comment:
1.

It is better to avoid copying of such list of images:
- use define
- or use single tests instantiation (see dnn/perf/perf_net.cpp):
```
PERF_TEST_P_(QRCode, detect)
{
...
}
PERF_TEST_P_(QRCode, decode)
{
...
}

INSTANTIATE_TEST_CASE_P(/*nothing*/, QRCode,
    ::testing::Values(
        ...
    )
);
```

2. Perhaps we should add performance tests for "negative" cases too (some images without qrcode (lena.png) + including random noise).